### PR TITLE
Locally build site and transfer it to deployer

### PR DESCRIPTION
**Pull request by Itxaka**
Ansible template is a bit slower, especially when doing local to
remote as it needs to check if the files exists, then check if
its different from the computed one and then template+transfer it
to the remote node.

In my local tests building the site from the templates could take from
30 minutes on a good connection to almost an hour in a connection with
a high latency (~1s).

Instead, lets just build the whole site in localhost under the workspace
and then transfer it on one go to the remote machine.

Currently I clean up the local copy of the site to keep everything
clean, but potentially we could leave those files up for ease of
checking what the site looks like without having to log into the
deployer. But this can come later, currently it just cleans up
everything at the end so we dont polute the workspace more than its
needed.

This results in an astonishing speed increase, taking only about 40s
in my high latency scenario (~1s). No downsides that I can see in this
approach

### CODE REVIEW

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml [hunk 1]
```diff
@@ -47,38 +53,48 @@
...
+    dest: "{{ site_path }}"
+    recursive: yes
+
+- name: Remove local site
+  file:
+    path: "{{ local_temp_path }}"
+    state: absent
+
 - name: Create pod scale profile
-  become: yes
```
Comment (jsuchome) [inline playbooks/roles/airship-deploy-ucp/tasks/main.yml]: Why can you remove `become` here? It does not seem relevant to the other checks

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 - name: Set site path
   set_fact:
-    site_path: "{{ upstream_repos_clone_folder }}/airship/treasuremap/site/{{ socok8s_site_name }}"
+    site_path: "{{ upstream_repos_clone_folder }}/airship/treasuremap/site/{{ socok8s_site_name }}/"
+    local_temp_path: "{{ socok8s_workspace }}/{{ socok8s_site_name }}/"
```
Comment (jsuchome) [inline playbooks/roles/airship-deploy-ucp/tasks/main.yml]: I think _local_ and _temp_ are bit redundant. What about `tmp_site_path`, `local_site_path` or something like that?

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml (hunk 1 above)
Comment (Itxaka) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: good point, I did miss this one

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml (hunk 2 above)
Comment (Itxaka) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: local its to indicate that its local to the ansible run, so its on local-host

I did not want to have any confusion between loca as in locahost and local as in local to the remote node

Comment (jsuchome) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: So you can just exclude `local_` completely and leave `tmp_site_path`?

Anyway, this is just me nitpicking, feel free to ignore

Comment (Itxaka) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: changed it to workspace_site_path so its clearer...maybe?

File: playbooks/roles/airship-deploy-ucp/tasks/main.yml (hunk 1 above)
Comment (Itxaka) [reply playbooks/roles/airship-deploy-ucp/tasks/main.yml]: fixed!
